crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -156,14 +169,17 @@ fn hmac_update<B: Board>(mut call: SchedulerCall<B, api::hmac_update::Sig>) {
     let memory = scheduler.applet.store.memory();
     let result = try {
         let data = memory.get(*data, *length)?;
-        match scheduler.applet.hashes.get_mut(*id as usize)? {
+        let context = match scheduler.applet.hashes.take(*id as usize)? {
             #[cfg(feature = "board-api-crypto-hmac-sha256")]
-            HashContext::HmacSha256(context) => context.update(data),
+            HashContext::HmacSha256(hmac) => HmacApi(hmac).update(data),
             #[cfg(feature = "board-api-crypto-hmac-sha384")]
-            HashContext::HmacSha384(context) => context.update(data),
+            HashContext::HmacSha384(hmac) => HmacApi(hmac).update(data),
             _ => trap_use!(data),
+        };
+        match context {
+            Ok(()) => Ok(()),
+            Err(error) => Err(error),
```

Comment:
@ia0 Without these lines, the compiler asked me to explicitly write out the type of result as it couldn't figure it out. There is probably a more elegant way.

Reply:
I can't reproduce this problem. Maybe it has been fixed in the meantime.